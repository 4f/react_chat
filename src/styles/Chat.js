export default theme => ({
  chatLayout: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    paddingTop: '64px',
    height: '100%',
    width: '100%',
    overflow: 'hidden'
  }
})