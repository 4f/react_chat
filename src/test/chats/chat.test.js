/* eslint-env jest */
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Chat from 'components/chats/chat';

jest.mock('components/chats/chat/messages', () => () => 'Messages');
jest.mock('components/chats/chat/input', () => () => 'inputMessage');

const mockProps = {
  user: {
    _id: 'ura',
    firstName: 'Name',
    lastName: 'Surname',
    username: 'username',
    isMember: true,
    isCreator: true,
    isInChat: true,
  },
  chat: {
    _id: '13',
    title: 'My Chat',
    messages: [
      {
        chatId: '13',
        content: 'mmm',
        sender: {},
        createdAt: '2018-03-26T11:11:11.100Z',
      },
    ],
  },
  join: jest.fn(),
  send: jest.fn(),
};

describe('<Chat />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <MemoryRouter>
        <Chat {...mockProps} />
      </MemoryRouter>,
      div,
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly ', () => {
    const tree = renderer
      .create(<MemoryRouter>
        <Chat {...mockProps} />
              </MemoryRouter>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
