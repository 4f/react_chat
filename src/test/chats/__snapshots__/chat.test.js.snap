// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chat /> renders correctly  1`] = `
<main
  className="Chat-chatLayout-11"
>
  Messages
  inputMessage
</main>
`;
